web/angular-wallet/src/app/main/transactions/transaction-table/transaction-table.component.ts

```diff
@@ -31,11 +44,24 @@ export class TransactionTableComponent implements OnInit {
   }
 
   public getTransactionNameFromType(transaction: Transaction): string {
-    return this.utilService.getTransactionNameFromType(transaction, this.account);
+    return this.utilService.translateTransactionType(transaction, this.account);
   }
 
   public isOwnAccount(address: string): boolean {
-    return address != undefined && address == this.account.accountRS;
-}
+    return address && address === this.account.accountRS;
+  }
+
+  public getAmount(transaction: Transaction): number {
 
+    if (this.isOwnAccount(transaction.senderRS)) {
+      return -this.convertNQTStringToNumber(transaction.amountNQT);
+    }
+
+    let amountNQT = transaction.amountNQT;
+    if (this.isMultiOutPayment(transaction)) {
+      const recipientAmounts = getRecipientsFromMultiOutPayment(transaction);
+      amountNQT = recipientAmounts.filter(ra => ra.recipient === this.account.account)[0].amountNQT;
```

Comment:
might be unsafe to assume `[0]` here

Reply:
I thought the same...theoretically, a multiout transaction should not allow same accounts. But better to code more defensive...